### PR TITLE
feat: message context menu + reactions

### PR DESCRIPTION
**Pull request by thierryskoda**
Here's a quick loom explaining what's missing and some patterns I've used for the `message-context-menu`
https://www.loom.com/share/71de204c9e7448ab80f3dd63caba8960

## Summary by CodeRabbit

- **New Features**
	- Introduced a context menu for messages, allowing users to react, reply, and perform other actions directly from the message interface.
	- Added emoji picker functionality for selecting reactions in the message context menu.
	- Enhanced handling of message reactions with improved visual feedback and organization.
	- Added a debug border style for attachment previews, aiding in visual debugging.
	- Introduced new components for managing context menus and reactions, improving user interaction.

- **Bug Fixes**
	- Improved error handling for message context actions and emoji selections.

- **Refactor**
	- Streamlined the `ChatMessageActions` component by removing unnecessary context menu logic.
	- Updated the `MessageReactions` component for better state management and clarity.
	- Refined the `MessageLayout` component to enhance gesture handling for message interactions.
	- Restructured the `V3Message` component for improved type safety and clarity.
	- Enhanced the `MessageTimestamp` and `MessageDateChange` components for better state responsiveness.

- **Chores**
	- Removed deprecated components and utilities related to message reactions and context menus.
	- Cleaned up import statements and optimized code structure for better maintainability.

### CODE REVIEW

File: queries/useConversationMessages.ts [hunk 1]
```diff
@@ -141,37 +143,68 @@ export const getConversationMessages = (
...
+              ],
+            },
+            byReactionContent: {
+              ...existingReactions.byReactionContent,
+              [reactionContentString]: [
+                ...(existingReactions.byReactionContent[
+                  reactionContentString
+                ] || []),
+                senderAddress,
+              ],
```
[the remark this answers is not in the document]
Comment (technoplato) [reply queries/useConversationMessages.ts:199]: OH SHIT CODE RABBIT FOUND THE BUG!!!

https://www.loom.com/share/bbf1bf4f6d384c6f8896e3494b380022

[the remark this answers is not in the document]
Comment (thierryskoda) [reply queries/useConversationMessages.ts]: WOW thank you RABBITT 🐰

File: components/Chat/Message/MessageReactions/MessageReactions.types.ts [hunk 2]
```diff
@@ -38,5 +38,5 @@ export type ReactorDetails = {
   address: string;
   userName?: string;
   avatar?: string;
-  reactionTime: number;
+  // reactionTime: number; // TODO
```
Comment (alexrisch) [inline components/Chat/Message/MessageReactions/MessageReactions.types.ts:41]: Is this still todo?

Comment (thierryskoda) [reply components/Chat/Message/MessageReactions/MessageReactions.types.ts:41]: No you're right removed it. At least for now it seems we're not doing anything with showing reaction timestamp

**Pull request merged**